tests/ui/coherence/with-negative-coherence-issue-119381.rs

```diff
@@ -0,0 +1,16 @@
+#![feature(with_negative_coherence)]
```

Comment:
The title of this issue doesn't really explain why it's an ICE. Specifically, there's no explanation for why we're encountering an inference variable in the type of a constant inside of a param-env, which is why this ICE occurs.